flink-dist/pom.xml

```diff
@@ -140,19 +140,54 @@ under the License.
 			</exclusions>
 		</dependency>
 
+		<!--
+			The following dependencies are packaged in 'plugin/'
+			The scope of these dependencies needs to be 'provided' so that
+			they are not included into the 'flink-dist' uber jar.
+		-->
 		<!-- Default file system support. The Hadoop and MapR dependencies -->
 		<!--       are optional, so not being added to the dist jar        -->
 
 		<dependency>
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-hadoop-fs</artifactId>
 			<version>${project.version}</version>
+			<scope>provided</scope>
```

Comment:
changing this may break plenty of existing user-setups that assume they only have to put hadoop on the classpath to be able to talk to hdfs.

Reply:
um... Actually, IMO, the merge-order of this PR should follow the order of subtasks under the umbrella issue about the plugin architecture (FLINK-11952). We should make sure the plugins work fine with the whole plugin architecture. There still is an opening issue which is being implemented by @pnowojski. cc @StefanRRichter

Reply:
I don't believe the merge order to be relevant here. None of the other subtasks are about re-introducing the hadoop/mapr compatibility code into flink-dist or having them on the classpath by default; as such my point still stands.

Conceptually I'm in favor of having these in the plugin directory as an opt-in feature, but nevertheless, we may break plenty of setups.

Reply:
That is, unless all plugins are in /plugins by default and https://issues.apache.org/jira/browse/FLINK-12143 aims to ship everything in that directory by default.

Reply:
@zentol Maybe I did not describe my thought clearly, I just want to say this PR is just an auxiliary work, it's final state will be decided by other works, design, back compatibility and so on. I think the who architecture is still in the exploration and testing stage.

I know that `flink-hadoop-fs` and `flink-mapr-fs` is been packaged into `flink-dist` before. And I know this PR is not in the merge-ready state. I just opened it, when the prior work is done, we will discuss how to deal with these two filesystems. There may be two possibilities :

* follow the plugin architecture totally, there may other work should be done as you said
* follow the plugin architecture partially, then these two modules should be packaged into `dist` like before, but others will be shipped into `plugin` dir